infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -138,26 +140,13 @@ public ManagementServer(ServerContext serverContext) {
             recovered = true;
         }
 
-        if ((Boolean) opts.get("--single")) {
-            String localAddress = opts.get("--address") + ":" + opts.get("<port>");
-
-            Layout singleLayout = new Layout(
-                    Collections.singletonList(localAddress),
-                    Collections.singletonList(localAddress),
-                    Collections.singletonList(new Layout.LayoutSegment(
-                            Layout.ReplicationMode.CHAIN_REPLICATION,
-                            0L,
-                            -1L,
-                            Collections.singletonList(
-                                    new Layout.LayoutStripe(
-                                            Collections.singletonList(localAddress)
-                                    )
-                            )
-                    )),
-                    0L
-            );
-
-            safeUpdateLayout(singleLayout);
+        synchronized (this.serverContext) {
+            if ((Boolean) opts.get("--single")) {
+                if (serverContext.getCurrentLayout() == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)